orderbook/src/main.rs

```diff
@@ -222,12 +200,8 @@ async fn main() {
         registry,
         metrics.clone(),
     );
-    let maintenance_task = task::spawn(orderbook_maintenance(
-        orderbook,
-        database.clone(),
-        settlement_contract,
-        current_block_stream,
-    ));
+    let maintenance_task =
+        task::spawn(current_block_stream.run_maintenance_on_new_block(service_maintainer));
```

Comment:
I'd prefer if instead of putting logic about maintenance into the current_block_stream component, we would instead pass the current_block_stream into the ServiceMaintenance maintenance component and do the stream processing there.

The ServiceMaintenance already know about the maintenance logic and I'd find it more natural to say this component work on current block stream than having the current block stream also know how to drive a maintenance component.

Reply:
Good suggestion.